views/js/controller/creator/helpers/changeTracker.js

```diff
@@ -0,0 +1,259 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 Open Assessment Technologies SA ;
+ */
+/**
+ * Track the change within the testCreator
+ * @author Juan Luis Gutierrez Dos Santos <juanluis.gutierrezdossantos@taotesting.com>
+ */
+define([
+    'jquery',
+    'lodash',
+    'i18n',
+    'lib/uuid',
+    'core/eventifier',
+    'ui/dialog'
+], function (
+    $,
+    _,
+    __,
+    uuid,
+    eventifier,
+    dialog
+) {
+    'use strict';
+
+    /**
+     * The messages asking to save
+     */
+    const messages = {
+        preview: __('The item needs to be saved before it can be previewed'),
+        leave: __('The item has unsaved changes, are you sure you want to leave ?'),
+        exit: __('The item has unsaved changes, would you like to save it ?')
+    };
+
+    /**
+     *
+     * @param {HTMLElement} container
+     * @param {testCreator} testCreator
+     * @param {String} [wrapperSelector]
+     * @returns {changeTracker}
+     * @fires stylechange when the item's style changed
+     */
+    function changeTrackerFactory(container, testCreator, wrapperSelector = 'body') {
+        let changeTracker;
+
+        // internal namespace for global registered events
+        const eventNS = `.ct-${uuid(8, 16)}`;
+
+        // keep the value of the item before changes
+        let originalItem;
```

Comment:
change `originalItem` to `originallTest`
and other places where `item` to `test`